cli/src/commands/simplify_parents.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashSet;
+
+use itertools::Itertools;
+use jj_lib::backend::BackendResult;
+use jj_lib::revset::RevsetExpression;
+use jj_lib::rewrite::CommitRewriter;
+use jj_lib::settings::UserSettings;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::RevisionArg;
+use crate::command_error::CommandError;
+use crate::ui::Ui;
+
+/// Simplify parent edges for the specified revision(s).
+///
+/// Removes all redundant parent edges going into any of the specified
+/// revisions. By definition, this has no effect on any commit contents and
+/// should not cause any working copy changes or conflicts.
+///
+/// In other words, for all (A, B, C) where A has (B, C) as parents and C is an
+/// ancestor of B, A will be rewritten to have only B as a parent instead of
+/// B+C.
+#[derive(clap::Args, Clone, Debug)]
+#[command(group = clap::ArgGroup::new("revision-args").multiple(true).required(true))]
+pub(crate) struct SimplifyParentsArgs {
+    /// Simplify specified revision(s) together with their trees of descendants
+    /// (can be repeated)
+    #[arg(long, short, group = "revision-args")]
+    sources: Vec<RevisionArg>,
+    /// Simplify specified revision(s) (can be repeated)
+    #[arg(long, short, group = "revision-args")]
+    revisions: Vec<RevisionArg>,
+}
+
+pub(crate) fn cmd_simplify_parents(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &SimplifyParentsArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let revs = RevsetExpression::descendants(
+        workspace_command
+            .parse_union_revsets(ui, &args.sources)?
+            .expression(),
+    )
+    .union(
+        workspace_command
+            .parse_union_revsets(ui, &args.revisions)?
```

Comment:
(No action needed) For "revisions", `rebase` currently uses plural and `parse_union_revsets`, so I think things are correct as they are.